test/view_component/view_component_test.rb

```diff
@@ -933,6 +933,26 @@ def test_each_component_has_a_different_lock
     assert_not_equal(MyComponent.compiler.__vc_compiler_lock, AnotherComponent.compiler.__vc_compiler_lock)
   end
 
+  def test_compilation_in_blocking_mode
+    ViewComponent::CompileCache.cache.delete(MyComponent)
+    MyComponent.compiler.mode = :blocking
+
+    render_inline(MyComponent.new)
+
+    assert_selector("div", text: "hello,world!")
+  end
+
+  def test_compilation_in_non_blocking_mode
+    ViewComponent::CompileCache.cache.delete(MyComponent)
+    MyComponent.compiler.mode = :non_blocking
+
+    render_inline(MyComponent.new)
+
+    assert_selector("div", text: "hello,world!")
+
+    MyComponent.compiler.mode = :blocking
```

Comment:
Probably a good idea to put this in an `ensure` so that a failure means other tests don't run in non-blocking mode.

Might also be good to provide a support method to this effect, assuming we'll need wider testing of this mode:

```ruby
def with_compiler_mode(mode, component)
  previous_mode = component.compiler.mode
  yield(component)
ensure
  component.compiler.mode = previous_mode
end